### PR TITLE
Rename classes folder to api

### PR DESCRIPTION
**Pull request by madmiraal**
To better reflect that the folder contains the Rebel Engine API ReST files, this PR renames the `classes` folder to `api`.

In addition, it updates the documentation to reflect the name change along with the corresponding understanding that it's an API. Furthermore, it includes updates the documentation to reflect the changes introduced by RebelToolbox/RebelEngine#26.

### CODE REVIEW

File: contributing/code_style_guidelines.rst [hunk 1]
```diff
@@ -342,7 +342,7 @@ Don't repeat what the code says in a comment. Explain the *why* rather than *how
 You can use Javadoc-style comments above function or macro definitions. It's
 recommended to use Javadoc-style comments *only* for methods which are not
 exposed to scripting. This is because exposed methods should be documented in
-the :ref:`class reference XML <doc_updating_the_class_reference>`
+the :ref:`API Documentation <doc_updating_the_api_documentaton>`
```
Comment (DanielaOrtner) [inline contributing/code_style_guidelines.rst]: I believe there is a typo in the last word of the line: "documentaton"

File: contributing/contributing_to_the_documentation.rst [hunk 2]
```diff
@@ -16,12 +16,12 @@ Modifying those pages in a pull request triggers a rebuild of the online documen
 .. seealso:: For details on Git usage and the pull request workflow, please
              refer to the :ref:`doc_pr_workflow` page.
 
-.. warning:: The class reference's source files are in the `Rebel Engine
+.. warning:: The API documenation source files are in the `Rebel Engine
```
Comment (DanielaOrtner) [inline contributing/contributing_to_the_documentation.rst]: Should the word "documenation" be "documentation" instead?

File: contributing/contributing_to_the_documentation.rst [hunk 3]
```diff
@@ -16,12 +16,12 @@ Modifying those pages in a pull request triggers a rebuild of the online documen
...
              repository <https://github.com/RebelToolbox/RebelEngine>`_. We generate
-             the :ref:`Rebel API <toc-class-ref>` section of this documentation
+             the :ref:`Rebel API <toc-api>` section of this documentation
              from them. If you want to update the description of a class, its
              methods, or properties, read
-             :ref:`doc_updating_the_class_reference`.
+             :ref:`doc_updating_the_api_documentaton`.
```
Comment (DanielaOrtner) [inline contributing/contributing_to_the_documentation.rst]: I believe there is a typo in the last word of the line: "documentaton"

File: contributing/contributing_to_the_documentation.rst [hunk 4]
```diff
@@ -53,10 +53,10 @@ Editing existing pages
 
 To edit an existing page, locate its ``.rst`` source file and open it in your
 favorite text editor. You can then commit the changes, push them to your fork,
-and make a pull request. **Note that the pages in** ``classes/`` **should not be
+and make a pull request. **Note that the pages in** ``api/`` **should not be
 edited here.** They are automatically generated from Rebel Engine's `XML class
-reference <https://github.com/RebelToolbox/RebelEngine/tree/main/doc/classes>`__.
-See :ref:`doc_updating_the_class_reference` for details.
+reference <https://github.com/RebelToolbox/RebelEngine/tree/main/docs>`__.
+See :ref:`doc_updating_the_api_documentaton` for details.
```
Comment (DanielaOrtner) [inline contributing/contributing_to_the_documentation.rst]: I believe there is a typo in the word: "documentaton"

File: contributing/documentation_guidelines.rst [hunk 5]
```diff
@@ -23,18 +23,18 @@ documentation.
...
              refer to the :ref:`doc_pr_workflow` page.
 
-.. warning:: The class reference's source files are in the `Rebel Engine repository
+.. warning:: The API documentation's source files are in the `Rebel Engine repository
              <https://github.com/RebelToolbox/RebelEngine>`_. We generate the :ref:`Rebel API
-             <toc-class-ref>` section of this documentation from them. If you want to update the
+             <toc-api>` section of this documentation from them. If you want to update the
              description of a class, its methods, or properties, read
-             :ref:`doc_updating_the_class_reference`.
+             :ref:`doc_updating_the_api_documentaton`.
```
Comment (DanielaOrtner) [inline contributing/documentation_guidelines.rst]: I believe there is a typo in the last word of the line: "documentaton"

File: contributing/documentation_guidelines.rst [hunk 6]
```diff
@@ -23,18 +23,18 @@ documentation.
...
 
-.. warning:: If you want to edit the **API reference**, please note that it
+.. warning:: If you want to edit the **Rebel Engine API**, please note that it
              should *not* be done in the Rebel Documenation repository. Instead, you
-             should edit the ``doc/classes/*`` XML files of Rebel Engine's
+             should edit the XML files in the ``docs/`` folder of Rebel Engine's
              repository. These files are then later used to generate the
              in-editor documentation as well as the API reference of the
-             online docs. Read more here: :ref:`doc_updating_the_class_reference`.
+             online docs. Read more here: :ref:`doc_updating_the_api_documentaton`.
```
Comment (DanielaOrtner) [inline contributing/documentation_guidelines.rst]: I believe there is a typo in the last word of the line: "documentaton"

File: contributing/index.rst [hunk 7]
```diff
@@ -53,20 +53,20 @@ Writing the manual
    contributing_to_the_documentation
    building_the_manual
 
-Class reference guides
-~~~~~~~~~~~~~~~~~~~~~~
+API Documentaiton
```
Comment (DanielaOrtner) [inline contributing/index.rst]: Maybe this should be "Documentation" instead of "Documentaiton"?

File: contributing/ways_to_contribute.rst [hunk 8]
```diff
@@ -180,13 +180,13 @@ Contributing to the documentation
...
+- **The API documentation.** This is the documentation for the complete Rebel Engine API
   as exposed to GDScript and the other scripting languages. It can be consulted
-  offline, directly in Rebel Engine's code editor, or online at :ref:`Rebel Engine API <toc-class-ref>`.
-  To contribute to the class reference, you have to edit the
+  offline, directly in Rebel Engine's code editor, or online at :ref:`Rebel Engine API <toc-api>`.
+  To contribute to the API documentation, you have to edit the
   XML file corresponding to the class and make a pull request.
-  See :ref:`doc_updating_the_class_reference` and
-  :ref:`doc_class_reference_writing_guidelines` for more details.
+  See :ref:`doc_updating_the_api_documentaton` and
```
Comment (DanielaOrtner) [inline contributing/ways_to_contribute.rst]: I believe there is a typo in the word: "documentaton"

File: development/cpp/custom_modules_in_cpp.rst [hunk 9]
```diff
@@ -470,32 +460,29 @@ main ``doc/classes`` directory.
...
 
-Now if you go to the ``RebelEngine/modules/summator/doc_classes`` folder, you will see
-that it contains a ``Summator.xml`` file, or any other classes, that you referenced
-in your ``get_doc_classes`` function.
+Now if you go to the ``RebelEngine/modules/summator/docs`` folder, you will see
+that it contains a ``Summator.xml`` file, or any other classes that you referenced
+in your ``get_classes`` function.
 
-Edit the file(s) following :ref:`doc_updating_the_class_reference` and recompile the engine.
+Edit the file(s) following :ref:`doc_updating_the_api_documentaton` and recompile the engine.
```
Comment (DanielaOrtner) [inline development/cpp/custom_modules_in_cpp.rst]: I believe there is a typo in the word: "documentaton"

File: tutorials/networking/http_request_class.rst [hunk 10]
```diff
@@ -71,9 +71,9 @@ Below is all the code we need to make it work. The URL points to an online API m
 
 With this, you should see ``(hello:world)`` printed on the console; hello being a key, and world being a value, both of them strings.
 
-For more information on parsing JSON, see the class references for :ref:`JSON <class_JSON>` and :ref:`JSONParseResult <class_JSONParseResult>`.
+For more information on parsing JSON, see the API documentations for :ref:`JSON <class_JSON>` and :ref:`JSONParseResult <class_JSONParseResult>`.
```
Comment (DanielaOrtner) [inline tutorials/networking/http_request_class.rst]: Should this really be the plural "documentations"?

File: tutorials/ui/bbcode_in_richtextlabel.rst [hunk 11]
```diff
@@ -16,7 +16,7 @@ a :ref:`class_RichTextLabel`.
 It has a built-in API for generating the markup, but can also parse a BBCode.
 
 Note that the BBCode tags can also be used, to some extent, in the
-:ref:`XML source of the class reference <doc_updating_the_class_reference>`.
+:ref:`XML source of the API documentation <doc_updating_the_api_documentaton>`.
```
Comment (DanielaOrtner) [inline tutorials/ui/bbcode_in_richtextlabel.rst]: I believe there is a typo in the word: "documentaton"

File: about/introduction.rst [hunk 12]
```diff
@@ -83,10 +83,10 @@ relatively intuitive:
 - The :ref:`sec-contributing` section gives information related to contributing to
   engine development, how to report bugs,
   help with the documentation, etc.
-- Finally, the :ref:`sec-class-ref` is the documentation of the Rebel Engine API,
+- Finally, the :ref:`sec-api` is the documentation of the Rebel Engine API,
   which is also available directly within the engine's script editor. It is
   generated automatically from a file in the main source repository, therefore
   the generated files of the documentation are not meant to be modified. See
-  :ref:`doc_updating_the_class_reference` for details.
+  :ref:`doc_updating_the_api_documentaton` for details.
```
Comment (DanielaOrtner) [inline about/introduction.rst]: I believe there is a typo in the word: "documentaton"

**Review by DanielaOrtner — CHANGES_REQUESTED** [review]
Could you please check the feedback? 

Thank you

**Comment by madmiraal** [comment]
Good spot! We clearly need to work out why the spell checker didn't pick these up.

**Comment by madmiraal** [comment]
Corrected all the incorrect spellings of documentation.

I'll submit a separate PR to address the problem with the spell checker not working.

**Comment by madmiraal** [comment]
Updated to include changes to the Rebel Engine API introduced by RebelToolbox/RebelEngine#25 and RebelToolbox/RebelEngine#26.

**Review by DanielaOrtner — APPROVED** [review]
Thank you for checking the spelling issues and fixing them.
All looks good now :)

**Pull request merged**